server/src/game/map/index.ts

```diff
@@ -392,40 +398,48 @@ export class Map {
       if (!overwrite) return;
       tile.owner?.removeTile(coords);
       if (tile.owner.civID) {
-        tile.setVisibility(tile.owner.civID, false);
+        tile.setVisibility(tile.owner.getDomainID(), false);
       }
     }
     tile.owner = owner;
     if (owner.civID) {
-      tile.setVisibility(owner.civID, true);
+      tile.setVisibility(tile.owner.getDomainID(), true);
     }
     owner.addTile(coords);
   }
 
-  private getTileDataByCiv(civID: number, tile: Tile): TileData | null {
-    if (!tile.discoveredBy[civID]) {
-      return null;
-    }
+  private getTileDataByDomainIDs(domainIDs: DomainID[], tile: Tile): TileData | null {
```

Comment:
I feel like this might be easier to read if it was written with the use of `.some(...)`? Not sure, though... just a thought.

Reply:
Maybe it would? I do like how I'm able to use `continue` here though, so unless you can think of something specific I think I'll keep it like this.